src/Helpers/CompositeHelper.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Bonnier\WP\ContentHub\Editor\Helpers;
+
+use Bonnier\WP\ContentHub\Editor\Models\WpAttachment;
+use Bonnier\VideoHelper;
+
+class CompositeHelper
+{
+    /**
+     * Composite constructor.
+     */
+    public function __construct()
+    {
+        add_action('save_post', [$this ,'videoTeaserImage']);
+    }
+
+    public function videoTeaserImage($postId)
+    {
+        if (wp_is_post_revision($postId) || !have_rows('composite_content', $postId)) {
+            return;
+        }
+
+        while (have_rows('composite_content', $postId)) {
```

Comment:
I dont like while loops, as there is always a chance they might loop infinitely and crash, so unless they are absolutely necessary i think we should avoid them, also in this loop your are making multiple queries to the database rather than just using 
``` php
$videoWithTeaser =collect(get_field('composite_content'))->first(function($content){
    return $content['acf_fc_layout] !== 'video' && $content['teaser_image'] ;
});
if($videoWithTeaser) {
$teaserImagefile = VideoHelper::getLeadImageFile($videoWithTeaser['embed_url'], 'https://bonnier-publications-danmark.23video.com');
// do the rest of your magic
}
```

Reply:
I do hate while loops too .. but this is the recommended way of fetching ACF data(flexible content), as they do it out of the box, so instead of re-inventing the wheel it's faster and easier to just use their pre-made functions for such a use. 

But nice catch tho, I just found out that I am doing a free DB call, which I don't need, so instead am just fetching the teaser_image checkbox and make a validation then do my magic. If not valid then just continue without any DB call.

I also added a break to exit the while loop once we do our magic ..

https://www.advancedcustomfields.com/resources/have_rows/
https://www.advancedcustomfields.com/resources/flexible-content/